leetcodePython/Tree/tree_543.py

```diff
@@ -0,0 +1,32 @@
+from typing import *
+
+
+class TreeNode:
+    def __init__(self, val=0, left=None, right=None):
+        self.val = val
+        self.left = left
+        self.right = right
+
+
+class Solution:
+    _diameter: int
+
+    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
+        self._diameter = 0
+
+        def get_max_height_to_parent(current: Optional[TreeNode]) -> int:
+            if not current:
+                return 0
+            left_height = get_max_height_to_parent(current.left)
+            right_height = get_max_height_to_parent(current.right)
+            self._diameter = max(self._diameter, left_height + right_height)
```

Comment:
The height we get is the max distance from parameter node's leaves to parameter node's parent